pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/kubelet.go

```diff
@@ -101,6 +102,7 @@ func KubeletConfigParametersFromCoreV1beta1KubeletConfig(kubeletConfig *gardenco
 		out.KubeReserved = reservedFromKubeletConfig(kubeletConfig.KubeReserved)
 		out.MaxPods = kubeletConfig.MaxPods
 		out.PodPidsLimit = kubeletConfig.PodPIDsLimit
+		out.ProtectKernelDefaults = kubeletConfig.ProtectKernelDefaults
```

Comment:
This flag is available only after Kubernetes version 1.26, Should we not check if the `ProtectKernelDefaults` value is set and kubernetes version is <=1.25 then it should be discarded? Otherwise, kubelet can throw error of unknown flag.

Reply:
This field is present in previous versions as well. See 1.17 [ref](https://github.com/kubernetes/kubernetes/blob/f3abc15296f3a3f54e4ee42e830c61047b13895f/staging/src/k8s.io/kubelet/config/v1beta1/types.go#L626). We just start defaulting it to `true` for 1.26 to prevent modifying existing clusters without upgrading them.

Reply:
What about doing it with Kubernetes 1.25, 1.25 is yet not rolled out on landscapes.